src/main/java/br/com/deveficiente/nossomercadolivreapi/usuario/UsuariosController.java

```diff
@@ -0,0 +1,31 @@
+package br.com.deveficiente.nossomercadolivreapi.usuario;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+
+@RestController
+@RequestMapping("/api/usuarios")
+public class UsuariosController {
+
+    @Autowired
+    private UsuarioRepository usuarioRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @PostMapping
+    public ResponseEntity<Void> cria(@RequestBody @Valid  UsuarioRequest usuarioRequest) {
```

Comment:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Aqui você pode retornar void direto. Eu sempre sou a favor de usar os defaults do framework até que os defaults não sejam suficientes.

Reply:
Alberto, neste ponto eu fiquei em dúvida pois o retorno que "pedem" para usar geralmente é o status code 201 - created e uma URL para o recurso recém criado mas na descrição da funcionalidade está pedindo para retornar 200 - OK.

Como posso escolher qual implementar, sempre pergunto ao "cliente" da aplicação?

Reply:
Sempre pergunta sim :). Aqui, neste contexto, pode mandar um 200 e está tudo certo.